integration-tests/keycloak-authorization/src/test/java/io/quarkus/it/keycloak/DynamicTenantConfigPolicyEnforcerTest.java

```diff
@@ -0,0 +1,250 @@
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+
+@QuarkusTest
+@QuarkusTestResource(KeycloakLifecycleManager.class)
+@TestProfile(DynamicTenantConfigPolicyEnforcerTest.DynamicTenantConfigResolverProfile.class)
+public class DynamicTenantConfigPolicyEnforcerTest extends AbstractPolicyEnforcerTest {
+
+    @Inject
+    KeycloakPolicyEnforcerConfig enforcerConfig;
+
+    public static class DynamicTenantConfigResolverProfile implements QuarkusTestProfile {
+        @Override
+        public String getConfigProfile() {
+            return "dynamic-config-resolver";
+        }
+    }
+
+    @Test
+    public void testDynamicConfigPermissionScopes() {
+        // 'jdoe' has scope 'read' and 'read' is required
+        assureGetPath("/api/permission/scopes/dynamic-way", 200, getAccessToken("jdoe"), "read");
+        assureGetPath("//api/permission/scopes/dynamic-way", 200, getAccessToken("jdoe"), "read");
+
+        // 'jdoe' has scope 'read' while 'write' is required
+        assureGetPath("/api/permission/scopes/dynamic-way-denied", 403, getAccessToken("jdoe"), null);
+        assureGetPath("//api/permission/scopes/dynamic-way-denied", 403, getAccessToken("jdoe"), null);
+    }
+
+    @Test
+    public void testDynamicConfigUserHasAdminRoleServiceTenant() {
+        assureGetPath("/dynamic-permission-tenant", 403, getAccessToken("alice"), null);
+        assureGetPath("//dynamic-permission-tenant", 403, getAccessToken("alice"), null);
+
+        assureGetPath("/dynamic-permission-tenant", 403, getAccessToken("jdoe"), null);
+        assureGetPath("//dynamic-permission-tenant", 403, getAccessToken("jdoe"), null);
+
+        assureGetPath("/dynamic-permission-tenant", 200, getAccessToken("admin"), "Permission Resource Tenant");
+        assureGetPath("//dynamic-permission-tenant", 200, getAccessToken("admin"), "Permission Resource Tenant");
+    }
+
+    @Test
+    public void testKeycloakPolicyEnforcerTenantConfigBuilder() {
+        assertBuilderPopulatedWithDefaultValues();
+        assertEveryConfigPropertyCanBeSet();
+        assertTenantConfigEnhanced(enforcerConfig.namedTenants().get("api-permission-tenant"));
+        assertBuilderShortcuts();
+    }
+
+    private static void assertBuilderPopulatedWithDefaultValues() {
+        var config = KeycloakPolicyEnforcerTenantConfig.builder().build();
+        assertEquals(20, config.connectionPoolSize());
+        assertTrue(config.policyEnforcer().lazyLoadPaths());
+        assertEquals(ENFORCING, config.policyEnforcer().enforcementMode());
+        assertTrue(config.policyEnforcer().paths().isEmpty());
+        assertFalse(config.policyEnforcer().httpMethodAsScope());
+        assertEquals(30000, config.policyEnforcer().pathCache().lifespan());
+        assertEquals(1000, config.policyEnforcer().pathCache().maxEntries());
+        assertTrue(config.policyEnforcer().claimInformationPoint().simpleConfig().isEmpty());
```

Comment:
`simpleConfig` and `complexConfig` should not be part of the Config builder API

Reply:
it is not, here the `config` is an interface annotated with `@ConfigMapping`. These lines assert that build config builder contains expected values.

Reply:
> `simpleConfig` and `complexConfig` should not be part of the Config builder API

they are not, so closing this comment thread as resolved